src/Factory/ParcelTrackingUrlFactory.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace Invertus\dpdBaltics\Factory;
+
+use Invertus\dpdBaltics\Config\Config;
+use Invertus\dpdBaltics\Provider\CurrentCountryProvider;
+use Invertus\dpdBaltics\Util\CountryUtility;
+
+class ParcelTrackingUrlFactory
+{
+    /**
+     * @var CurrentCountryProvider
+     */
+    private $countryProvider;
+
+    public function __construct(CurrentCountryProvider $countryProvider)
+    {
+        $this->countryProvider = $countryProvider;
+    }
+    public function createTrackingUrl($cart, $parcelNumber)
+    {
+
+        $currentCountry = $this->countryProvider->getCurrentCountryIsoCode($cart);
+
+        if (in_array($currentCountry, Config::VALID_TRACKING_URL_COUNTRIES, true)) {
+            $countryIsoCode = CountryUtility::toLowerCase($currentCountry);
+
+            return "https://www.dpdgroup.com/{$countryIsoCode}/mydpd/tmp/basicsearch?lang={$countryIsoCode}&parcel_id={$parcelNumber}";
+        }
+
+        return "https://www.dpdgroup.com/lt/mydpd/tmp/basicsearch?lang=en&parcel_id={$parcelNumber}";
```

Comment:
there's LT inside the link and i'm pretty sure we support more than LT. Maybe the url should be an input somewhere in the module configuration?

Reply:
@tomjas1997 they daclared these links, so i just figured I can make them dynamic way
![image](https://user-images.githubusercontent.com/28574753/129553613-e8791ca2-b1ed-4a46-9eb4-24006f3220ce.png)

Reply:
Yeah, you did one part of it correctly. /www.dpdgroup.com/lt/ though is not correct. 

Lt should be countryISO just as lang parameter is.

Reply:
Ok  checked the link again, seems kinda weird that dpd would have lt inside their link if they use it for multiple countries. 

But if it is correct then resolving the conversation.